cors.go

```diff
@@ -0,0 +1,85 @@
+type CORS struct {
+	// AllowedMethods that scripts running in the browser are permitted to use.
+	//
+	// To support cross-domain requests with the protocols supported by Connect,
+	// these headers fields must be included in the preflight response header
+	// Access-Control-Allow-Methods.
+	AllowedMethods []string
+
+	// AllowedHeader fields that scripts running in the browser are permitted
+	// to send.
+	//
+	// To support cross-domain requests with the protocols supported by Connect,
+	// these field names must be included in the preflight response header
+	// Access-Control-Allow-Headers.
+	//
+	// Make sure to include any application-specific headers your browser client
+	// may send.
+	AllowedHeaders []string
+
+	// ExposedHeader fields that scripts running the browser are permitted
+	// to see.
+	//
+	// To support cross-domain requests with the protocols supported by Connect,
+	// these field names must be included in header Access-Control-Expose-Headers
+	// of the actual response.
+	//
+	// Make sure to include any application-specific headers your browser client
+	// should see. If your application uses trailers, they will be sent as header
+	// fields with a `Trailer-` prefix for Connect unary RPCs. Ensure to
+	// include them as well if you want them to be visible in all supported
+	// protocols.
+	ExposedHeaders []string
+}
+
+// CORSDefault returns a new CORS config object with the default required
+// configuration for Connect and supported protocols.
+//
+// Make sure to include application-specific headers that your application
+// uses as well in addition to these default headers.
+func CORSDefault() CORS {
+	return CORS{
+		AllowedMethods: []string{http.MethodGet, http.MethodPost},
+		AllowedHeaders: []string{
+			headerContentType,
+			connectHeaderProtocolVersion,
+			connectHeaderTimeout,
+			connectUnaryHeaderCompression,           // Unused in web browsers, but added for future-proofing
+			connectUnaryHeaderAcceptCompression,     // Unused in web browsers, but added for future-proofing
+			connectStreamingHeaderCompression,       // Unused in web browsers, but added for future-proofing
+			connectStreamingHeaderAcceptCompression, // Unused in web browsers, but added for future-proofing
+			grpcHeaderTimeout,                       // Used for gRPC and gRPC-web
+			grpcHeaderMessageType,                   // Unused in web browsers, but added for future-proofing
+			grpcWebHeaderXGrpcWeb,                   // Used for gRPC-web
+			grpcWebHeaderXUserAgent,                 // Used for gRPC-web
+		},
+		ExposedHeaders: []string{
+			connectUnaryHeaderCompression,     // Unused in web browsers, but added for future-proofing
+			connectStreamingHeaderCompression, // Unused in web browsers, but added for future-proofing
+			grpcHeaderStatus,                  // Required for gRPC-web
+			grpcHeaderMessage,                 // Required for gRPC-web
+			grpcHeaderDetails,                 // Error details in gRPC, gRPC-web
```

Comment:
In general, I'd remove references to gRPC from the comments and focus on gRPC-Web. CORS is only relevant for browser support, and gRPC is unusable from browsers.